components/match_table/commons/match_table.lua

```diff
@@ -0,0 +1,752 @@
+		timeZone = record.extradata.timezoneid or UTC,
+		liquipediatier = record.liquipediatier,
+		liquipediatiertype = record.liquipediatiertype,
+		displayName = String.nilIfEmpty(record.tournament) or record.pagename:gsub('_', ' '),
+		icon = String.nilIfEmpty(record.icon),
+		iconDark = String.nilIfEmpty(record.icondark),
+		pageName = String.nilIfEmpty(record.parent) or String.nilIfEmpty(record.pagename),
+		vods = self:vodsFromRecord(record),
+		type = record.type,
+		result = result,
+		game = record.game,
+	}
+end
+
+---@param record table
+---@return {index: number, link: string}[]
+function MatchTable:vodsFromRecord(record)
+	local vods = {}
+	if String.nilIfEmpty(record.vod) then
+		vods = {{index = 0, link = record.vod}}
+	end
+
+	Array.forEach(record.match2games, function(game, gameIndex)
+		if String.nilIfEmpty(game.vod) then
+			table.insert(vods, {link = game.vod, index = gameIndex})
+		end
+	end)
+
+	return vods
+end
+
+---@param record table
+---@return MatchTableMatchResult?
+function MatchTable:resultFromRecord(record)
+	if #record.match2opponents ~= 2 then
+		return self:resultFromNonStandardRecord(record)
+	end
+
+	local aliases = self.config.aliases
+	local countGames = false
+
+	local foundInAlias = function(opponentRecord)
+		if aliases[opponentRecord.name] then
+			countGames = true
+			return true
+		end
+		return self.config.mode == PLAYER_MODE and Array.any(opponentRecord.match2players, function(player)
+			return aliases[player.name] or false
+		end)
+	end
+
+	local winner = tonumber(record.winner)
+	local indexes
+	if foundInAlias(record.match2opponents[1]) then
+		indexes = {1, 2}
+	elseif foundInAlias(record.match2opponents[2]) then
+		indexes = {2, 1}
+		winner = winner == 2 and 1 or winner == 1 and 2 or winner
+	else
+		mw.ext.TeamLiquidIntegration.add_category('MatchesTables with invalid matches')
+		mw.logObject(record)
+		return
+	end
+
+	local result = {
+		opponent = record.match2opponents[indexes[1]],
+		vs = record.match2opponents[indexes[2]],
+		winner = winner,
+		resultType = record.resultType,
+		countGames = countGames,
+	}
+
+	return result
+end
+
+---overwritable for wikis that have BR/FFA matches
+---@param record table
+---@return table?
+function MatchTable:resultFromNonStandardRecord(record)
+end
+
+---@return {matches: {w: number, d: number, l: number}, games: {w: number, d: number, l: number}}
+function MatchTable:statsFromMatches()
+	local totalMatches = {w = 0, d = 0, l = 0}
+	local totalGames = {w = 0, d = 0, l = 0}
+
+	local nonNegative = function(value)
+		return math.max(tonumber(value) or 0, 0)
+	end
+
+	Array.forEach(self.matches, function(match)
+		---TODO: decide if we want default wins to count towards this
+		if match.result.resultType == DRAW then
+			totalMatches.d = totalMatches.d + 1
+		elseif match.result.winner == 1 then
+			totalMatches.w = totalMatches.w + 1
+		elseif match.result.winner == 2 then
+			totalMatches.l = totalMatches.l + 1
+		end
```

Comment:
opinions?

Reply:
Default wins means BYE opponents here or also W-FF results against real opponents?

Reply:
didn't think of byes
will probably exclude them via the query conditions (`resulttype::!np`)

this section is about `W-FF`, `W-DQ`, etc

Reply:
I think we should include these, unless we also mention the amount of them specifically. It would be weird if d+w+l is not equal to #totalMatches

Reply:
so lets just vote on this
👍 --> count default wins
👎 --> exclude default wins

default win = `[[resulttype::default]]` and `[[resulttype::!np]]`